air/src/execution_step/instructions/canon_utils/mod.rs

```diff
@@ -65,15 +102,24 @@ pub(crate) fn handle_unseen_canon(
     if exec_ctx.run_parameters.current_peer_id.as_str() != peer_id {
         exec_ctx.make_subgraph_incomplete();
         exec_ctx.next_peer_pks.push(peer_id);
-        //this branch is executed only when
-        //  this canon instruction executes for the first time
-        //  a peer is different from one set in peer_id of a this canon instruction
-        //
-        // the former means that there was no canon associated state in data, and the latter means
-        // that it can't be obtained on this peer, so it's intended not to call meet_canon_end here.
-        return Ok(());
+
+        trace_ctx.meet_canon_end(CanonResult::request_sent_by(
+            exec_ctx.run_parameters.current_peer_id.clone(),
+        ));
```

Comment:
Looks like if the RSB canon result will be created on a separate line, then it should be two not three lines in total.